### ISSUE TITLE
Stucked tunnel

### ISSUE THREAD

**Issue by privorka** [localtunnel/localtunnel#391]
My tunnel got stucked and even when killing the process the tunnel seems to be still "active".
The subdomain doesn't want to get initialized so I guess it's still in use even though the tunel process on my side has been terminated. The url itself returns 504.
Is there a way to make the subdomain available? Or is there any timeout that will do it automatically?

**Comment by TheBoroer** [comment]
Hey,
That seems to be an issue when the client experiences a weird internet connection issue or the client doesn't disconnect property.
I haven't had the time to look into a fix or introduce a timeout or keepalive yet but i'll get to it soon I promise

**Comment by privorka** [comment]
Yeah, that's most likely the issue here. I was messing around with DHCP settings on my router and probably caused it to get stuck in this weird state.
A keepalive would be nice.
Or I was thinking about generating a simple token on startup that would allow you to kill the tunnel from "outside" if it gets stuck (just an idea - not sure how "doable" that would be :)).

**Comment by blackgearit** [comment]
It happens to me too. I've a FWA internet access, i'm behind a provider's NAT. When client disconnect for some reason, at the next attempt the requested subdomain is not assigned again, and this is a problem.

**Comment by JannikZed** [comment]
Happens very often to me :/ I think a session should time-out way faster, so that you can take it often..

**Comment by privorka** [comment]
I have to agree that it started to be a real pain. It fails to me in average twice a day and I had to switch back to ngrok.
I love the service though and if you get to fixing it, I will be the first one to switch back :).

I originally switched from ngrok because when I had to kill the ngork tunnel, it generated a new url which led to changing it in all the 3rd party services (15minutes work each time it changes).
Btw. I also tried to setting up the server which I was successful after some time but the problem is that it won't work on heroku, so I got back to the beginning :D.

**Comment by maganius** [comment]
I have the same issue, in my case have dynamic ip and when it change or the internet get disconnected with the tunnel active the subdomain get stucked.

**Comment by allanvb** [comment]
any updates on this ? it still happening. Thank you in advance

**Comment by RickCastle2018** [comment]
They're already 3 same problem issues here and there's a question on Stackoverflow. @TheBoroer please fix it 😥🙏

**Comment by dorqa95** [comment]
I am facing with the same issue, please look after it if that's possible. Thanks for your effort!

**Comment by lonewolfaka9** [comment]
+1
